### DISCUSSION TITLE
Help with the Expectation-Maximization (EM_CUDA) algorithm

### DISCUSSION THREAD

**Discussion by DSWanZo** [astra-toolbox/astra-toolbox#443] [Q&A] [answered]
Hi,
I am using the Matlab version of ASTRA and I have a problem with the EM-CUDA algorithm. 
I have tried the example in the documentation (below) and it gives me a black image.

```matlab
V_exact = phantom(256);
proj_geom = astra_create_proj_geom('parallel', 1.0, 256, linspace2(0,pi,180));
vol_geom = astra_create_vol_geom(256,256);

[sinogram_id, sinogram] = astra_create_sino_cuda(V_exact, proj_geom, vol_geom);

recon_id = astra_mex_data2d('create', '-vol', vol_geom, 0);
cfg = astra_struct('EM_CUDA');
cfg.ProjectionDataId = sinogram_id;
cfg.ReconstructionDataId = recon_id;
em_id = astra_mex_algorithm('create', cfg);
astra_mex_algorithm('iterate', em_id, 15);
V = astra_mex_data2d('get', recon_id);
``` 

As I understand it, I think this is because the algorithm needs an initial reconstruction but I can't see where to input the initial reconstruction. 

Thanks a lot !

**Comment by askorikov** [comment, marked as answer]
Hi! Oops, it seems the docs are incorrect. Because EM algorithm applies multiplicative updates, you need to initialize reconstruction with non-zero numbers (typically all ones) -- otherwise nothing will get updated during iterations. I have just fixed that [in the documentation](https://astra-toolbox.com/docs/algs/EM_CUDA.html). Thanks for reporting!

**Reply by DSWanZo** [reply]
Thank you !
